virtualizarr/manifests/manifest.py

```diff
@@ -507,7 +503,10 @@ def validate_chunk_keys(chunk_keys: Iterable[ChunkKey]):
 
 def get_chunk_grid_shape(chunk_keys: Iterable[ChunkKey]) -> tuple[int, ...]:
     # find max chunk index along each dimension
-    zipped_indices = zip(*[split(key) for key in chunk_keys])
+    if list(chunk_keys) == ["c"]:
+        # Scalar array, cannot be split
+        return ()
+    zipped_indices = zip(*[parse_manifest_index(key) for key in chunk_keys])
```

Comment:
In case `chunk_keys` cannot be iterated over more than once:

```suggestion
    chunk_keys = tuple(chunk_keys)
    
    if chunk_keys == ("c",):
        # Scalar array, cannot be split
        return ()
    zipped_indices = zip(*map(parse_manifest_index, chunk_keys))
```